projects/transformers/README.md

```diff
@@ -13,13 +13,15 @@ In progress, current results:
 | bert_700k |          79.77 |          75.25 |  47.73 | 83.91/84.22 | 88.27/83.59 |  91.51 | 90.37/86.90 | 62.11 |  91.44 | 86.88/86.70 |  39.06 |  |
 | bert_100k |          75.71 |          72.51 |  40.98 | 78.26/78.65 | 84.05/77.86 |  87.74 | 89.12/85.25 |  58.2 |  88.31 | 83.90/83.80 |  46.88 | 8.619 |
 | sparse_v1_100k |          72.67 |          67.91 |  22.43 | 77.25/77.96 | 85.37/78.82 |  84.65 | 88.23/84.32 | 58.32 |  87.27 | 82.79/82.65 |  29.84 | 10.461 |
-| sparse_v2_100k |  |  |  |  |  |  |  |  |  |  |  |  |
+| sparse_v2_100k |          68.41 |          67.06 |  15.96 | 72.25/73.11 | 80.50/71.35 |  81.23 | 85.78/81.01 | 58.98 |  82.64 | 76.53/76.40 |  56.25 | 16.696 |
+| sparse_v3_100k |          68.25 |          66.92 |  16.35 | 72.00/72.32 | 80.42/70.83 |  80.31 | 85.04/80.49 | 58.98 |  82.52 | 76.96/77.60 |  56.25 | 22.81 |
```

Comment:
It is also strange wnli results are so high in v2 and v3 if we haven't made any changes to the finetuning regime - what could cause that sudden increase?

Reply:
Could the high WNLI accuracy just be variance?

Reply:
I'm gonna run WNLI again and see if I get something high again.

Reply:
Per the perplexity comment: It's possible that since perplexity involves an exponent (as in $2^{H(P, Q)}), the difference between values could be more exaggerated as it increases. So maybe a jump in perplexity from 8 to 10 is roughly the same as a jump from 16 to 22. That it is to say, it's hard to compare perplexities at face value if they are not on a linear scale. My understanding of how they are calculated is still limited, so maybe I'm wrong on this. I'll try and think more about how to interpret their values.

Reply:
Good point on the perplexity not being on linear scale. 

> I'm gonna run WNLI again and see if I get something high again.

No need to run again. Remember all values are saved, and the reduction is only performed at the export_results script. You can inspect by loading the pickled class and printing all results, and that should tell whether it is a matter of variance or not. If that is the case, the means should be more coherent than the maxes.

Reply:
You can even modify the TaskResults to have new reduction methods, including for example mean with margin of error. The original intention in making it a class was to add these new functionalities as required.

Reply:
Although I recognize it would be cleaner to save it as raw human readable data and have reductions methods in a separate module that handles the data. (todo)

Reply:
On the perplexity being on an exponential scale - if it fits our intuition better, we can track log(ppl) along with ppl.

Reply:
I think we already do. It's the loss I believe. From our `run_utils.py` code:
```
perplexity = math.exp(eval_output["eval_loss"])
```

Reply:
That is strange. We are using regular cross entropy (negative loglikelihood). That means perplexity is the exponential of the cross entropy loss? It doesn't match my understanding - we should definitely look into this at some point. 

Also, not clear to me how perplexity is calculated for masked language model, it is not a very direct application. We talked about this when we were discussing perplexity.